pkg/controller/channel/controller.go

```diff
@@ -305,16 +305,16 @@ func (c *Controller) syncHandler(key string) error {
 
 	// Sync Service derived from the Channel
 	service, serviceErr = c.syncChannelService(channel)
-	if err != nil {
```

Comment:
Note that to reach this point, err must be nil, as it is last set on line 290. The 'then' portion of 291-300 will always return if err is non-nil.